tests/common/nats.rs

```diff
@@ -2,64 +2,99 @@ use std::env;
 
 use anyhow::{Context, Result};
 use async_nats::connection::State;
+use tokio::net::TcpStream;
 use tokio::process::Command;
 use tokio::time::{sleep, timeout, Duration};
 use url::Url;
 
 use super::{free_port, tempdir, BackgroundServer};
 
-pub async fn start_nats() -> Result<(BackgroundServer, Url, async_nats::Client)> {
+pub async fn start_nats(
+    config: Option<String>,
+    return_client: bool,
+) -> Result<(BackgroundServer, Url, Option<async_nats::Client>)> {
```

Comment:
I might just be dumb, was there a scenario where we couldn't return the client that motivated the optional return type?